Specs/Scene/GlobeSurfaceTileProviderSpec.js

```diff
@@ -88,41 +82,35 @@ defineSuite([
     function updateUntilDone(globe) {
         // update until the load queue is empty.
         return pollToPromise(function() {
-            globe.beginFrame(frameState);
-            globe.update(frameState);
-            globe.endFrame(frameState);
+            scene.renderForSpecs();
             return globe._surface.tileProvider.ready && !defined(globe._surface._tileLoadQueue.head) && globe._surface._debug.tilesWaitingForChildren === 0;
         });
     }
 
-    function switchTo2D() {
-        frameState.mode = SceneMode.SCENE2D;
-        var frustum = new OrthographicFrustum();
-        frustum.right = Ellipsoid.WGS84.maximumRadius * Math.PI;
-        frustum.left = -frustum.right;
-        frustum.top = frustum.right;
-        frustum.bottom = -frustum.top;
-        frameState.camera.frustum = frustum;
-        frameState.camera.update(frameState.mode);
-        frameState.camera.setView({ destination : new Rectangle(0.0001, 0.0001, 0.0030, 0.0030) });
+    function switchViewMode(mode, projection) {
+        scene.mode = mode;
+        scene.frameState.mapProjection = projection;
+        scene.camera.update(scene.mode);
+        scene.camera.setView({
+            destination : new Rectangle(0.0001, 0.0001, 0.0030, 0.0030)
+        });
     }
 
     beforeAll(function() {
-        context = createContext();
+        scene = createScene();
+        scene.frameState.scene3DOnly = false;
     });
 
     afterAll(function() {
-        context.destroyForSpecs();
+        scene.destroyForSpecs();
     });
 
     beforeEach(function() {
-        frameState = createFrameState(context);
-        globe = new Globe();
-        surface = globe._surface;
+        scene.globe = new Globe();
     });
 
     afterEach(function() {
-        globe.destroy();
+        scene.imageryLayers.removeAll();
```

Comment:
If you look in [Scene.js](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Scene/Scene.js#L690), the old globe is automatically destroyed when the new one is set